src/dctest/core.cljs

```diff
@@ -177,87 +179,119 @@ Options:
            (when error {:error error})
            (when signal {:signal signal}))))
 
-(defn get-expect-error [context expect]
-  (when-not (expr/read-eval context expect)
-    {:message (str "Expectation failure: " expect)
-     :debug (expr/explain-refs context expect)}))
+(defn update-any-with-context [{:keys [context results]} k f]
```

Comment:
The `k` vs `ks` thing was one of the things I thrashed on quite a bit as I worked through options here.  Ultimately I went this direction for two reasons:  1) all of the functions here take and return context+results essentially, so I thought it might be ok that results was not explicit (and perhaps even weird if it were explicit), and 2) I didn't update error messages yet (I had implemented it a few times, but again the diff was larger and harder to process), but I would like good, localized error messages, ex: "Error evaluating 'if': ...".  It seemed weird to me if the function that created that message ignored the first key in `ks` if we pass `[:results :if]`.

But I do see what you're saying. Do you think a better name for the function would help, or is this just misconceived?

As for `update-any-with-context`, yeah naming was hard here again.  This function is like `update` except it carries/lifts the context along as the first argument.  Also, it looks at the type of the thing it's updating (is it a string? vector? map?), and handles those cases slightly differently.  We don't mind if our values are in a "container" like vector or map, we just want to update them all.  The `any` is trying to differentiate this aspect from the normal `update` behavior, I guess (think "any type"?).  Suggestions definitely welcome.

Reply:
Nevermind - this can be cleaner, I'll push changes. Thanks!

Reply:
Pushed a change with a more generic `update-in*` (better names welcome)